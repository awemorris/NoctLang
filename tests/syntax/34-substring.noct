func main() {
    var s = "ABC123あいう";

    print(s);
    print("s.length = " + s.length);
    print("s.substring(0, 0) = " + s->substring(0, 0));
    print("s.substring(0, 1) = " + s->substring(0, 1));
    print("s.substring(0, 2) = " + s->substring(0, 2));
    print("s.substring(0, 3) = " + s->substring(0, 3));
    print("s.substring(0, 4) = " + s->substring(0, 4));
    print("s.substring(0, 5) = " + s->substring(0, 5));
    print("s.substring(0, 6) = " + s->substring(0, 6));
    print("s.substring(0, 7) = " + s->substring(0, 7));
    print("s.substring(0, 8) = " + s->substring(0, 8));
    print("s.substring(0, 9) = " + s->substring(0, 9));
    print("s.substring(0, 10) = " + s->substring(0, 10));
    print("s.substring(0, -1) = " + s->substring(0, -1));

    print("s.substring(1, 0) = " + s->substring(1, 0));
    print("s.substring(1, 1) = " + s->substring(1, 1));
    print("s.substring(1, 2) = " + s->substring(1, 2));
    print("s.substring(1, 3) = " + s->substring(1, 3));
    print("s.substring(1, 4) = " + s->substring(1, 4));
    print("s.substring(1, 5) = " + s->substring(1, 5));
    print("s.substring(1, 6) = " + s->substring(1, 6));
    print("s.substring(1, 7) = " + s->substring(1, 7));
    print("s.substring(1, 8) = " + s->substring(1, 8));
    print("s.substring(1, 9) = " + s->substring(1, 9));
    print("s.substring(1, 10) = " + s->substring(1, 10));
    print("s.substring(1, -1) = " + s->substring(1, -1));

    print("s.substring(8, 0) = " + s->substring(8, 0));
    print("s.substring(8, 1) = " + s->substring(8, 1));

    print("s.substring(9, 0) = " + s->substring(9, 0));
    print("s.substring(9, 1) = " + s->substring(9, 1));
}
