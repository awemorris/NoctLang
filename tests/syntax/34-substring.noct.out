ABC123あいう
s.length = 15
s.substring(0, 0) = 
s.substring(0, 1) = A
s.substring(0, 2) = AB
s.substring(0, 3) = ABC
s.substring(0, 4) = ABC1
s.substring(0, 5) = ABC12
s.substring(0, 6) = ABC123
s.substring(0, 7) = ABC123あ
s.substring(0, 8) = ABC123あい
s.substring(0, 9) = ABC123あいう
s.substring(0, 10) = ABC123あいう
s.substring(0, -1) = ABC123あいう
s.substring(1, 0) = 
s.substring(1, 1) = B
s.substring(1, 2) = BC
s.substring(1, 3) = BC1
s.substring(1, 4) = BC12
s.substring(1, 5) = BC123
s.substring(1, 6) = BC123あ
s.substring(1, 7) = BC123あい
s.substring(1, 8) = BC123あいう
s.substring(1, 9) = BC123あいう
s.substring(1, 10) = BC123あいう
s.substring(1, -1) = 
s.substring(8, 0) = 
s.substring(8, 1) = う
s.substring(9, 0) = 
s.substring(9, 1) = 
